File: GFG/LinkedList/rectra.cpp
//this code is to travers the linkedlist in 3 ways Loop,Recursion,Inbuilt function

#include<iostream>
#include"../ds.h"
using namespace std;

void nortra(Node<int>* ll){
    Node<int>* curr=ll;
    if(curr==NULL){
        cout<<"List is empty";
        return;
    }
    while(curr!=NULL){
        cout<<curr->data<<"->";
        curr=curr->next;
    }
    cout<<"None";
    return;
}
void rectra(Node<int>* ll){
    if (ll==NULL){
        return ;
    }
    Node<int>*curr=ll;
    
    rectra(curr->next);
    cout<<curr->data<<"->";
}


int main(){
    LinkedList<int> tl;
    for(int i=1;i<11;i++){
        tl.insert(i);
    }
    //first way using inbuilt function
    //tl.display();
    //cout<<endl;
    //nortra(tl.head);
    cout<<endl;
    rectra(tl.head);
    cout<<"None";
    return 0;

}

File: GFG/LinkedList/rectra.py
import ds

def nortra(ll):
    if ll is None:
        return 
    curr=ll
    while(curr!=None):
        print(curr.data,end=" ")
        curr=curr.next
    return

def rectra(ll):
    if(ll is None):
        return
    print(ll.data,end=" ")
    return rectra(ll.next)

tl=ds.LinkedList()
for i in range(1,11):
    tl.insert(i)

tl.display()
print("*")
nortra(tl.head)
print("*")
rectra(tl.head)